### PR TITLE
Recommend to disable logging to pihole.log on new installs

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [X] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**

Recommend to disable logging to pihole.log on new installs/during reconfigure. We do not use the log file since we introduced *FTL*DNS. It is still used for general (error) logging of the embedded `dnsmasq`, so it still makes sense to keep it in the debugger:
```
#log-queries
log-facility=/var/log/pihole.log
```

**How does this PR accomplish the above?:**

Change default selected option during the question and move the `(Recommended)` from `On` to `Off`.

**What documentation changes (if any) are needed to support this PR?:**

Probably none.

### CODE REVIEW

**Comment by Mcat12** [comment]
Holding on merge until we determine if really we want to change this.

**Comment by dschaper** [comment]
Is there some discussion that needs to happen for this?

**Comment by Mcat12** [comment]
Previous discussion brought up some issues around documentation and debugging: https://team.pi-hole.net/pi-hole/pl/rutrci1tsbrmtqoq77mhoopk4o

**Comment by jfb-pihole** [comment]
From a troubleshooting perspective (particularly the ability to tail the pihole log), logging should be enabled by default.  It puts very little wear on an SD card, the files are rotated and pruned regularly, and it is quite helpful when solving Pi-hole problems.

**Comment by dschaper** [comment]
I can see the argument for and against it. The troubleshooting process is actually quite a rarity, we just see it very often because that's what we do. To me it would make more sense to disable it by default and then toggle it on for debugging/troubleshooting when needed.

**Comment by jfb-pihole** [comment]
That adds an extra loop in the troubleshooting process.  As is, it is easy to glance at the bottom of a debug log and see the log entries.  Many times this helps solve the problem at the first crack.  I frequently have users check the log for entries of this type or that type, and without the log they don't have the information.  Even if they enable the log, it may take quite a bit of time to generate the data, and even then, the data might not be relevant to a past problem.

**Comment by PromoFaux** [comment]
This is just query logging, right? `pihole-FTL.log` will still remain in place?

**Comment by MichaIng** [comment]
Just to clarify: The `pihole.log` is the query log which matches the one from web UI exactly, right? So all this information about why this or that was (not) blocked, due to which rule etc, can be gathered from web UI, which is what users most likely check first anyway?

`pihole-FTL.log` contains the binary/dnsmasq log, which is relevant if process/service or DHCP/DNS fails and such, web UI shows red dot, so query log is not relevant in such case.

Did I get it right?
_________
So basically how I understand it, is that query logging is currently doubled: In web UI/database and to file. The question is whether it is a large benefit when users can check log file on top of web UI for query logs, if one serves better format (which can be adjusted). Also wasn't there a pihole command/option to print query logs from database to console, when needed?

**Comment by DL6ER** [comment]
You are correct in most places @MichaIng even if not did not interpret everything in the right way. Let me summarize the most important files and what the command you refer to does:

- `pihole.log` contains all queries and how they are replied to but also DHCP/DNS status messages
- `pihole-FTL.log` contains only very few information and is only interesting in case of debugging
- `pihole-FTL.db` (the database file) contains almost anything that is also in `pihole.log` (this is where information is dublicated)
- `pihole -t` as command shows the tail of `pihole.log` so it does not show anything when logging to the file is disabled

**Comment by MichaIng** [comment]
@DL6ER 
Many thanks for clarifying. Wouldn't it make sense to make `pihole -t` print database logs, as an console alternative for opening the web UI? It sounds pretty useless to me to have a command which does not more than `tail /var/log/pihole.log`? Currently it is mostly the same, but doing that change would remove another use case/need for `pihole.log`.

**Comment by DL6ER** [comment]
@MichaIng `pihole -t` is much more than a simple `tail`ing. It has some colorization onboard and highlights important bits in the logs. The database is only updated once per minute (users may have even set this to wider intervals) to reduce wearing out the storage medium. I don't think this is of much concern these days, however, the rumors that SD card are dying frequently cannot easily be extinct. As long as enough users are convinced that this is causing problem, we will take care to reduce these issues.

Due to this delay, `pihole -t` doesn't seem to be useful when sourcing the database. However, there is something coming up for the new web interface (projected Pi-hole v6.0), see https://github.com/pi-hole/web/pull/423
This is possible as FTL will keep the log in RAM from where they can be queries through the API.

**Comment by MichaIng** [comment]
@DL6ER 
Ah yeah as long as there is no database caching layer, this makes sense. Although it is as well an argument against file logging. `pihole.log` is the reason why we (until having it disabled by default) on DietPi needed to raise the default /var/log tmpfs size and still found cases where 50 MiB was reached within a single day. The basic issue, that a single additional log entry can lead to the whole size being rewritten to SDcard (when not having a tmpfs log dir), is exactly the same as with the database file, isn't it?

**Comment by MichaIng** [comment]
Just got reminded about this by a forum topic regarding an own RAM log implementation (not ours).

I have not enough insights, but to preserve debugging capabilities, probably `pihole -t` could be reworked to read `pihole-FTL.db` instead of `pihole.log`? If this is only written on a defined time interval, probably it is possible to have `pihole -t` read and print those logs from the originating source, from where it is written to `pihole-FTL.db` as well? This could even speed up the parsing compared to reading all logs from an on-disk file?

**Comment by DL6ER** [comment]
For Pi-hole v6.0 (or whatever the version featuring out own embedded HTTP server inside FTL will turn out to be), we already implemented a live streaming from FTL to the web interface and also `pihole -t` (see https://github.com/pi-hole/web/pull/423). I implemented this to (hopefully) finally remove any dependency on this file...

This implementation has the additional benefit of being a rolling memory, i.e., the user will not only receive all messages from point X in time where they start `pihole -t` but also some information from the past.

**Comment by MichaIng** [comment]
@DL6ER 
That sounds great. Okay then it does not make sense to develop some v5-only implementation now that will be obsolete with v6 of course.

So the question remains if reduced SDcard wear and duplicated logging is enough argument against debugging limitations. Both is minor I would say. For debugging worst case is an additional troubleshooting loop, and heavily used Pi-hole instances, where one can expect accordingly large `pihole.log` files, will or should anyway not run on little SBCs with low-quality SDcards.

**Comment by dschaper** [comment]
v6 has no known date at this time.

>So the question remains if reduced SDcard wear and duplicated logging is enough argument against debugging limitations.

No, not really.

**Comment by dschaper** [comment]
As long as people can use `pihole -t` to get a real-time view of the domains being seen then I'm okay. If it's on any kind of delay then I'm against it. Trying to find the domain that needs to be whitelisted when a site is not behaving is difficult enough as it is, adding in a delay between when the URL is visited in the browser and the display of the domains would cause any attempt to whitelist to become a very frustrating experience.

**Comment by MichaIng** [comment]
@dschaper 
Ah I forgot about `pihole -t` again. This is probably the strongest argument to keep things as they are, since `pihole -t` simply does not work without `pihole.log`, right? Not good to have a command option broken by default.

**Comment by DL6ER** [comment]
This may be revisited when v6.0 is there, because it will support tailing the log without storing any log physically on-disk (FTL maintains a FIFO buffer for dnsmasq log lines).
